src/autopas/tuning/selectors/TraversalSelector.h

```diff
@@ -68,12 +68,14 @@ class TraversalSelector {
    * @param traversalType
    * @param pairwiseFunctor
    * @param info
+   * @param useSorting Only for traversals that use the CellFunctor: If the CellFunctor should sort particles
    * @return Smartpointer to the traversal.
    */
   template <class PairwiseFunctor, DataLayoutOption::Value dataLayout, bool useNewton3>
   static std::unique_ptr<TraversalInterface> generateTraversal(TraversalOption traversalType,
                                                                PairwiseFunctor &pairwiseFunctor,
-                                                               const TraversalSelectorInfo &info);
+                                                               const TraversalSelectorInfo &info,
+                                                               const bool useSorting = true);
```

Comment:
I'd argue this should not be a constructor argument. Either remove it as suggested somewhere else or make it part of the `TraversalSelectorInfo` struct

Reply:
Here, I would definitely put it into the `TraversalSelectorInfo` struct.

Reply:
How about this idea: In the `CellFunctor`, give the functions `processCellAoS()`, `processCellPairAoSN3()` and `processCellPairAoSNoN3()` an additional argument `useSorting` and remove the member variable `_useSorting` from the `CellFunctor`. Give the `TraversalInterface` a protected member `_useSorting` (default to true) and a corresponding setter. The traversals that use a `CellFunctor` can then use the value of the inherited `_useSorting` when calling the process functions from `CellFunctor`.
If I leave the member variable `_useSorting` in the `CellFunctor` and write a setter in the `CellFunctor`, I see the following problem: I would have to write a virtual setter for `_useSorting` (or getter for `CellFunctor`) in the `TraversalInterface`, which would have to be implemented in every traversal.
The problem is, that in the `TraversalComparison` tests I have only access to a `TraversalInterface`, and in `CellFunctor` I don’t have access to the traversal that uses this `CellFunctor`.

If I put `useSorting` in `TraversalSelectorInfo`, I have to leave `useSorting` as a constructor argument for all traversals that use a `CellFunctor`.

Reply:
Do all Traversals that use the `CellFunctor` implement the same Class? Maybe something like `CellBasedTraversal`? Then you could put the getter/setter there and in the Test check if the Traversal object can be cast to that type?

Reply:
Then I would have to use `CellPairTraversal` to also cover the Octree traversals and DirectSum. However, there would then also be some traversals that implement this class but do not use `CellFunctor`.

Maybe a new interface `CellFunctorTraversal` or something like that?

Reply:
Can you please quickly put together the list of CellPairTraversals that do not use the CellFunctor and the list of all traversals that do use it? I don't have the full overview :sweat_smile:

Reply:
Here is a list of all traversals that implement `CellPairTraversal`
| Use CellFunctor    | Do not use CellFunctor |
|--------------------|------------------------|
| ds_sequential      | lc_c04_combined_SoA    |
| lc_c01             | vcl_c06                |
| lc_c04_HCP         | vcl_sliced_balanced    |
| lc_c04             | vcl_sliced_c02         |
| lc_c08             | vcl_sliced             |
| lc_c18             | vlc_c01                |
| lc_sliced_balanced | vlc_c18                |
| lc_sliced          | vlc_sliced             |
| lc_sliced_c02      | vlc_sliced_balanced    |
| lc_c01_combined_SoA | vlc_sliced_c02         |
| ot_01              | vlp_c01                |
| ot_18              | vlp_c18                |
|                    | vlp_sliced             |
|                    | vlp_sliced_balanced    |
|                    | vlp_sliced_c02         |

Reply:
I would still add the functionality to `CellPairTraversal`.  Imo, it's not really a problem that it just does nothing in case of vcl, vlc, vlp traversals.
I would definitely try to not make the traversal inheritance structure more complex just for the sorting parameter. (Because it will already become more convoluted with upcoming 3-body traversals :D)

Reply:
done in [8081014](https://github.com/AutoPas/AutoPas/pull/619/commits/80810142d34cf7872346976089a90e1753180bb7)